tab-oneipl/minutes/2021_12_16_Minutes.rst

```diff
@@ -0,0 +1,147 @@
+
+* Example with splitting pipeline. Two functions taking single input and
+  multiple outputs can be executed in parallel, since calls are asynchronous.
+
+* More complex pipeline with watermarking (slide 14) – user-provided kernels
+  like calculating overlay and blending. Some stages can produce output
+  required to host, so shared memory allocator shall be specified in this case.
+  If the image is pure device all intermediate output shall have device
+  allocator.
+
+* oneIPL Image Abstraction. Image can be potentially implemented over host,
+  shared, device and special texture memory. From implementation perspective
+  first three can be done via USM (SYCL2020), texture can be implemented
+  via sycl::image.
+
+* Memory model. oneapi::ipl::image class is basic data abstraction for image
+  data. oneIPL provides single abstraction over different memory types.
+  oneIPL supports different types of memory – device, host, shared and special
+  GPU texture (image) memory.
+
+* See more details on `oneIPL Architecture page`_
+
+
+oneIPL specification open discussion:
+
+* Robert Schneider, Siemens Healthineers: Should all API be used by HW
+  accelerated types?
+  Victor Getmanskiy, Intel: HW accelerated types are currently very limited
+  by SYCL standard images – only 4-channel images of limited data type are
+  supported. We are trying to extend image support to wider amount of
+  format-type combination in next SYCL standard. So the wider HW-accelerated
+  support would be possible in that case.
+
+* Robert Schneider, Siemens Healthineers: Are there query for HW feature like
+  format/type?
+  Victor Getmanskiy, Intel: Good question, we also see a gap here and consider
+  such extension to current SYCL images in future standards. Currently we can
+  query only if image/FP64/FP16 are supported or not.
+
+* Sergey Ivanov, OpenCV, Intel: You've said, the SYCL queue carries device
+  context - does all the objects need to belong to the same context?
+  Victor Getmanskiy, Intel: Yes, and specification recommends for each function
+  to have a runtime checks that queue and image objects belong to same context.
+  Checks can be disabled in implementation to avoid performance overhead.
+  sycl::queue and data pointer on slide 12 are specified in the image
+  constructor so implementation can determine, if the pointer and queue are
+  related to the same device.
+
+* Tim van der Horst, Philips: If the checks for the supported data types are
+  done before kernel, are they done in each call?
+  Victor Getmanskiy, Intel: Implementation can be done in a way to allow to
+  disable the checks to avoid performance overhead. In this case single initial
+  check is required in user Application.
+
+* Robert Schneider, Siemens Healthineers: Can the ipl::image be constructed
+  over sycl::image?
+  Victor Getmanskiy, Intel: Currently sycl::image 2020 is not supported by any
+  compiler, and SYCL image specification is improving, but in future based on
```

Comment:
SYCL image specification is being improved

Reply:
thanks, fixed